### PR TITLE
Add type definitions for `gauge` package.

### PR DESCRIPTION
**Pull request by brianloveswords**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

**Comment by mhegazy** [comment]
@brianloveswords please address the CI test failures.

File: types/gauge/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
+// Type definitions for gauge 2.7
+// Project: https://github.com/iarna/gauge
+// Definitions by: Brian J Brennan <https://github.com/brianloveswords>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+declare namespace Gauge {
+    interface Stream { }
```
Comment (andy-ms) [inline types/gauge/index.d.ts]: TypeScript uses a structural type system, so an empty interface is equivalent to the `{}` types, which accepts any value other than `null` or `undefined`.

File: types/gauge/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        isEnabled(): boolean;
+        pulse(subsection?: string): void;
+        setTemplate(template: Template): void;
+        setTheme(theme: Theme | ThemeOpts | string): void;
+        setThemeset(themes: ThemeSet): void;
+        show(sectionOrStatus?: string | StatusObject, completed?: number): void;
+    }
+}
+
+declare module "gauge" {
```
Comment (andy-ms) [inline types/gauge/index.d.ts]: Please use a separate file for each of these, e.g. `gauge/index.d.ts`, `gauge/themes.d.ts`, `gauge/theme-set.d.ts`.

**Comment by yuit** [comment]
Thanks for your contribution. This PR has unaddress comment/failing CI and cannot be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

**Pull request closed without merging**